string name
float32[] segment_x
float32[] segment_y